include/core/solid_objects_parameters.h

```diff
@@ -238,9 +264,12 @@ namespace Parameters
         verbosity = Verbosity::quiet;
 
       number_solids = prm.get_integer("number of solids");
-      for (unsigned int i_solid = 0; i_solid < number_solids; ++i_solid)
+      nitsche_solids.resize(number_solids);
+
+      if (number_solids > 0)
         {
-          nitsche_solids[i_solid]->parse_parameters(prm, i_solid);
+          for (unsigned int i_solid = 0; i_solid < number_solids; ++i_solid)
+            nitsche_solids[i_solid]->parse_parameters(prm, i_solid);
         }
```

Comment:
Why do you need an if around the for? The for loop won't do anything is the number of solid is 0.